static/app/components/dropdownMenuItemV2.tsx

```diff
@@ -126,171 +125,162 @@ type Props = {
    * this ref
    */
   submenuTriggerRef?: React.RefObject<HTMLLIElement>;
-} & WithRouterProps;
+};
 
 /**
  * A menu item with a label, optional details, leading and trailing elements.
  * Can also be used as a trigger button for a submenu. See:
  * https://react-spectrum.adobe.com/react-aria/useMenu.html
  */
-const MenuItem = withRouter(
-  ({
-    node,
-    isLastNode,
-    state,
-    onClose,
-    closeOnSelect,
-    isSubmenuTrigger = false,
-    submenuTriggerRef,
-    renderAs = 'li' as React.ElementType,
-    router,
-    ...submenuTriggerProps
-  }: Props) => {
-    const [isHovering, setIsHovering] = useState(false);
-    const ref = submenuTriggerRef ?? useRef(null);
-    const isDisabled = state.disabledKeys.has(node.key);
-    const isFocused = state.selectionManager.focusedKey === node.key;
-    const item = node.value;
+const MenuItem = ({
+  node,
+  isLastNode,
+  state,
+  onClose,
+  closeOnSelect,
+  isSubmenuTrigger = false,
+  submenuTriggerRef,
+  renderAs = 'li' as React.ElementType,
+  ...submenuTriggerProps
+}: Props) => {
+  const [isHovering, setIsHovering] = useState(false);
+  const ref = submenuTriggerRef ?? useRef(null);
+  const isDisabled = state.disabledKeys.has(node.key);
+  const isFocused = state.selectionManager.focusedKey === node.key;
+  const item = node.value;
+
+  const actionHandler = () => {
+    if (item.to) {
+      return;
+    }
+    if (isSubmenuTrigger) {
+      state.selectionManager.select(node.key);
+      return;
+    }
+    item.onAction?.(item.key);
+  };
 
-    const actionHandler = () => {
+  // Open submenu on hover
+  const {hoverProps} = useHover({onHoverChange: setIsHovering});
+  useEffect(() => {
+    if (isHovering && isFocused) {
       if (isSubmenuTrigger) {
         state.selectionManager.select(node.key);
         return;
       }
-      item.onAction?.(item.key);
-      item.to && router.push(item.to);
-    };
+      state.selectionManager.clearSelection();
+    }
+  }, [isHovering, isFocused]);
 
-    // Open submenu on hover
-    const {hoverProps} = useHover({onHoverChange: setIsHovering});
-    useEffect(() => {
-      if (isHovering && isFocused) {
-        if (isSubmenuTrigger) {
-          state.selectionManager.select(node.key);
-          return;
-        }
-        state.selectionManager.clearSelection();
+  // Open submenu on arrow right key press
+  const {keyboardProps} = useKeyboard({
+    onKeyDown: e => {
+      if (e.key === 'Enter' && item.to) {
+        const mouseEvent = new MouseEvent('click', {
+          ctrlKey: e.ctrlKey,
+          metaKey: e.metaKey,
+        });
+        ref.current?.querySelector(`${InnerWrap}`)?.dispatchEvent(mouseEvent);
+        onClose();
+        return;
       }
```

Comment:
if this item is a link, and if the user pressed `Enter`, then manually trigger the link